control_plane/attachment_service/src/service.rs

```diff
@@ -2819,28 +2783,18 @@ impl Service {
                     // No change
                 }
             };
-            node.availability = *availability;
+            node.set_availability(*availability);
```

Comment:
You could make `set_availability` return something to indicate a transition from offline to active or vice versa.
This would remove the need for `get_availability` which is an awkward part of the api imo.

Reply:
I've done this, and used it to refactor the calling function to avoid using bare bools for logic.  Makes things nicer.